impeller/compiler/shader_lib/impeller/prefix_sum.glsl

```diff
@@ -0,0 +1,38 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Parallel exclusive prefix sum computes the prefix in place in storage.
+// BLOCK_SIZE is the overall storage size while ident must be the global
+// x identifier.
+#define ExclusivePrefixSum(ident, storage, BLOCK_SIZE) \
```

Comment:
Why a macro and not a function?

Reply:
Needs to work over threadgroup shared memory which can't be passed as a ptr in glsl as far as I can tell

Reply:
Right, GLSL doesn't have pointers, you can't pass shared memory to a function defined in another file (it will try to do a copy)